### PR TITLE
Add documentation to automate sourcemaps upload

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Add documentation to automate sourcemaps upload

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

**Comment by urseberry** [comment]
https://datadoghq.atlassian.net/browse/DOCS-3903 created for docs team review

**Review by xgouchet — APPROVED** [review]

File: docs/crash_reporting.md [hunk 1]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
 npm install --save-dev @datadog/datadog-ci
 ```
 
+#### Automatically on each release build (React Native >= 0.69)
+
+Manually uploading your sourcemaps on every release build takes time and is prone to errors. We recommend automatically sending your sourcemaps every time you run a release build.
```
Comment (alai97) [inline docs/crash_reporting.md:77]:
```suggestion
Manually uploading your source maps on every release build takes time and is prone to errors. Datadog recommends automatically sending your source maps every time you run a release build.
```

File: docs/crash_reporting.md [hunk 2]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+Create a new script file named `datadog-sourcemaps.sh` at the root of your project, containing this:
```
Comment (alai97) [inline docs/crash_reporting.md:79]:
```suggestion
Create a script file named `datadog-sourcemaps.sh` at the root of your project containing the following:
```

File: docs/crash_reporting.md [hunk 3]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+```shell
+#!/bin/sh
+
+# If the build runs from XCode, we cannot use yarn.
+# Therefore we need to check first which yarn command is appropriate
```
Comment (alai97) [inline docs/crash_reporting.md:85]:
```suggestion
# Check first which yarn command is appropriate
```

File: docs/crash_reporting.md [hunk 4]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+REACT_NATIVE_XCODE="node_modules/react-native/scripts/react-native-xcode.sh"
+DATADOG_XCODE="$(echo $PACKAGE_MANAGER_BIN) datadog-ci react-native xcode"
+
+/bin/sh -c "$DATADOG_XCODE $REACT_NATIVE_XCODE"
+```
+
+This script finds the best way to run the `yarn datadog-ci react-native xcode` command:
+
+-   `yarn` can be used if you use a tool like [fastlane][9] or a service like [Bitrise][10] or [AppCenter][11] to build your app
+-   `/opt/homebrew/bin/node /opt/homebrew/bin/yarn` must be used on Mac if you run the release build from XCode directly
```
Comment (alai97) [inline docs/crash_reporting.md:105]:
```suggestion
- `yarn` can be used if you use a tool like [fastlane][9] or a service like [Bitrise][10] or [AppCenter][11] to build your app
- `/opt/homebrew/bin/node /opt/homebrew/bin/yarn` must be used on Mac if you run the release build from XCode directly
```

File: docs/crash_reporting.md [hunk 5]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+It then runs this command that will take care of uploading the sourcemaps with all the right parameters for you. See [datadog-ci documentation][12] for more information.
```
Comment (alai97) [inline docs/crash_reporting.md:107]:
```suggestion
It runs this command that takes care of uploading the source maps with all the correct parameters. For more information, see the [datadog-ci documentation][12].
```

File: docs/crash_reporting.md [hunk 6]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+Open your `.xcworkspace` with XCode, then select your project > Build Phases > Bundle React Native code and images. Edit the script to look like this:
```
Comment (alai97) [inline docs/crash_reporting.md:109]:
```suggestion
Open your `.xcworkspace` with XCode, then select your project > Build Phases > Bundle React Native code and images. Edit the script to look like the following:
```

File: docs/crash_reporting.md [hunk 7]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+For the upload to work, you need to provide your Datadog API key. If you use a command-line tool or an external service, you can specify it as a `DATADOG_API_KEY` environment variable. If you run the build from XCode, create a `datadog-ci.json` file at the root of your project containing the API key:
+
+```json
+{
+    "apiKey": "<YOUR_DATADOG_API_KEY>"
+}
+```
+
+You can also specify the Datadog site (e.g. `datadoghq.eu`) as a `DATADOG_SITE` environment variable, or as a `datadogSite` key in your `datadog-ci.json` file.
```
Comment (alai97) [inline docs/crash_reporting.md:130]:
```suggestion
You can also specify the Datadog site (such as `datadoghq.eu`) as a `DATADOG_SITE` environment variable, or as a `datadogSite` key in your `datadog-ci.json` file.
```

File: docs/crash_reporting.md [hunk 8]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+#### Automatically on each release build (React Native < 0.69)
+
+Open your `.xcworkspace` with XCode, then select your project > Build Phases > Bundle React Native code and images. Edit the script to look like this:
```
Comment (alai97) [inline docs/crash_reporting.md:134]:
```suggestion
Open your `.xcworkspace` with XCode, then select your project > Build Phases > Bundle React Native code and images. Edit the script to look like the following:
```

File: docs/crash_reporting.md [hunk 9]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+export SOURCEMAP_FILE=./build/main.jsbundle.map
+$(echo $PACKAGE_MANAGER_BIN datadog-ci react-native xcode)
+```
+
+This script finds the best way to run the `yarn datadog-ci react-native xcode` command:
+
+-   `yarn` can be used if you use a tool like [fastlane][9] or a service like [Bitrise][10] or [AppCenter][11] to build your app
+-   `/opt/homebrew/bin/node /opt/homebrew/bin/yarn` must be used on Mac if you run the release build from XCode directly
+
+It then runs this command that will take care of uploading the sourcemaps with all the right parameters for you. See [datadog-ci documentation][12] for more information.
```
Comment (alai97) [inline docs/crash_reporting.md:161]:
```suggestion
- `yarn` can be used if you use a tool like [fastlane][9] or a service like [Bitrise][10] or [AppCenter][11] to build your app
- `/opt/homebrew/bin/node /opt/homebrew/bin/yarn` must be used on Mac if you run the release build from XCode directly

It runs this command that takes care of uploading the source maps with all the correct parameters. For more information, see the [datadog-ci documentation][12].
```

File: docs/crash_reporting.md [hunk 10]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
+
+For the upload to work, you need to provide your Datadog API key. If you use a command-line tool or an external service, you can specify it as a `DATADOG_API_KEY` environment variable. If you run the build from XCode, create a `datadog-ci.json` file at the root of your project containing the API key:
+
+```json
+{
+    "apiKey": "<YOUR_DATADOG_API_KEY>"
+}
+```
+
+You can also specify the Datadog site (e.g. `datadoghq.eu`) as a `DATADOG_SITE` environment variable, or as a `datadogSite` key in your `datadog-ci.json` file.
```
Comment (alai97) [inline docs/crash_reporting.md:171]:
```suggestion
You can also specify the Datadog site (such as `datadoghq.eu`) as a `DATADOG_SITE` environment variable, or as a `datadogSite` key in your `datadog-ci.json` file.
```

File: docs/crash_reporting.md [hunk 11]
```diff
@@ -141,9 +239,27 @@ yarn datadog-ci react-native upload --platform ios --service $SERVICE --bundle $
 
 ### Upload JavaScript source maps on Android builds
 
+#### Automatically on each release build
+
+In your `android/app/build.gradle` file, add right after the `apply from: "../../node_modules/react-native/react.gradle"` line:
```
Comment (alai97) [inline docs/crash_reporting.md:244]:
```suggestion
In your `android/app/build.gradle` file, add the following after the `apply from: "../../node_modules/react-native/react.gradle"` line:
```

File: docs/crash_reporting.md [hunk 12]
```diff
@@ -141,9 +239,27 @@ yarn datadog-ci react-native upload --platform ios --service $SERVICE --bundle $
...
+
+For the upload to work, you need to provide your Datadog API key. You can specify it as a `DATADOG_API_KEY` environment variable, or create a `datadog-ci.json` file at the root of your project containing the API key:
+
+```json
+{
+    "apiKey": "<YOUR_DATADOG_API_KEY>"
+}
+```
+
+You can also specify the Datadog site (e.g. `datadoghq.eu`) as a `DATADOG_SITE` environment variable, or as a `datadogSite` key in your `datadog-ci.json` file.
```
Comment (alai97) [inline docs/crash_reporting.md:258]:
```suggestion
You can also specify the Datadog site (such as `datadoghq.eu`) as a `DATADOG_SITE` environment variable, or as a `datadogSite` key in your `datadog-ci.json` file.
```

**Review by alai97 — APPROVED** [review]
Looks good with the following copy suggestions!

File: docs/crash_reporting.md [hunk 13]
```diff
@@ -72,6 +72,104 @@ yarn add -D @datadog/datadog-ci
...
 ```
 
+#### Automatically on each release build (React Native >= 0.69)
+
+Manually uploading your sourcemaps on every release build takes time and is prone to errors. We recommend automatically sending your sourcemaps every time you run a release build.
+
+Create a new script file named `datadog-sourcemaps.sh` at the root of your project, containing this:
+
+```shell
+#!/bin/sh
```
Comment (0xnm) [inline docs/crash_reporting.md:82]: should we add `set -e` here the same way it appears in the script for pre-0.69 versions?

**Review by 0xnm — APPROVED** [review]
nice!

File: docs/crash_reporting.md (hunk 13 above)
Comment (louiszawadzki) [reply docs/crash_reporting.md:82]:
```suggestion
#!/bin/sh
set -e
```

File: docs/crash_reporting.md (hunk 3 above)
Comment (louiszawadzki) [reply docs/crash_reporting.md:85]:
```suggestion
# Check first which yarn executable is appropriate 
```

**Pull request merged**